qiskit/pulse/device_specification.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Specification of the device.
+"""
+import logging
+from typing import List
+
+from qiskit.pulse.qubit import Qubit
+from qiskit.pulse.channels import DriveChannel, ControlChannel, MeasureChannel
+from qiskit.pulse.channels import AcquireChannel, MemorySlot, RegisterSlot
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceSpecification:
+    """Implement a device specification, which is usually contructed from backend info."""
+
+    @classmethod
+    def create_from(cls, backend):
+        """
+        Create device specification with values in backend configuration.
+        Args:
+            backend(Backend):
+        """
+        config = backend.configuration()
+
+        # system size
+        n_qubit = config.n_qubits
+
+        # frequency information
+        qubit_lo_freqs = config.defaults['qubit_freq_est']
+        qubit_lo_range = config.qubit_lo_range
+        meas_lo_freqs = config.defaults['meas_freq_est']
+        meas_lo_range = config.meas_lo_range
+
+        # generate channels with assuming their numberings are aligned with qubits
+        drives = [DriveChannel(i, qubit_lo_freqs[i]) for i in range(n_qubit)]   # TODO: lo_ranges
```

Comment:
Here we still struggle with measurement los being tied to a specific `Schedule`?

Reply:
Yes. We still have the issue on los. I'm going to address it via another PR.